Mathlib/Probability/ProbabilityMassFunction/Integrals.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2023 Joachim Breitner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joachim Breitner
+-/
+import Mathlib.Probability.ProbabilityMassFunction.Basic
+import Mathlib.Probability.ProbabilityMassFunction.Constructions
+import Mathlib.MeasureTheory.Integral.Bochner
+
+/-!
+# Integrals with a measure derived from probability mass functions.
+
+This files connects `Pmf` with `integral`. The main result is that the integral (i.e. the expected
+value) with regard to a measure derived from a `Pmf` is a sum weighted by the `Pmf`.
+
+It also provides the expected value for specific probability mass functions.
+-/
+
+namespace Pmf
+
+open MeasureTheory BigOperators ENNReal
+
+theorem integral_eq_tsum' (α : Type _) [MeasurableSpace α] [MeasurableSingletonClass α] (p : Pmf α)
+  (f : α → ℝ) (hf : Integrable (fun a ↦ f a) (p.toMeasure)) :
+    ∫ a, f a ∂(p.toMeasure) = ∑' a, f a * (p a).toReal := calc
```

Comment:
```suggestion
variable {α : Type _}

theorem integral_eq_tsum' [MeasurableSpace α] [MeasurableSingletonClass α] (p : Pmf α)
    (f : α → ℝ) (hf : Integrable (fun a ↦ f a) (p.toMeasure)) :
      ∫ a, f a ∂(p.toMeasure) = ∑' a, f a * (p a).toReal := calc
```
Also, I think it would really make sense for the results in this file to be generalized to `f : α → β` for appropriate typeclasses on `β`. This would mean changing the statement to something like `∫ a, f a ∂(p.toMeasure) = ∑' a, (p a).toReal • f a`.

Reply:
I don't know the type class hierarchy well. Should I try with the type classes you used in the suggestion at `integrable_of_fintype`?